### PR TITLE
Update TBLogger docs

### PR DESCRIPTION
**Pull request by s-rog**
## What does this PR do?

Addresses some concerns about TBLogger hparams loggings docs. #2974

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by s-rog** [comment]
Previews here:
https://102900-178626720-gh.circle-artifacts.com/0/html/extensions/logging.html
https://102900-178626720-gh.circle-artifacts.com/0/html/extensions/generated/pytorch_lightning.loggers.TensorBoardLogger.html#pytorch_lightning.loggers.TensorBoardLogger

File: pytorch_lightning/loggers/tensorboard.py [hunk 1]
```diff
@@ -46,10 +46,13 @@ class TensorBoardLogger(LightningLoggerBase):
     preinstalled.
 
     Example:
-        >>> from pytorch_lightning import Trainer
-        >>> from pytorch_lightning.loggers import TensorBoardLogger
-        >>> logger = TensorBoardLogger("tb_logs", name="my_model")
-        >>> trainer = Trainer(logger=logger)
+
+    .. code-block:: python
+
```
Comment (carmocca) [inline pytorch_lightning/loggers/tensorboard.py:51]: I believe this is not necessary, `Example:` renders already

but haven't checked

```suggestion
```

Comment (s-rog) [reply pytorch_lightning/loggers/tensorboard.py:51]: Yes but I thought I'd keep it consistent with other loggers

File: pytorch_lightning/loggers/tensorboard.py [hunk 2]
```diff
@@ -46,10 +46,13 @@ class TensorBoardLogger(LightningLoggerBase):
     preinstalled.
 
     Example:
-        >>> from pytorch_lightning import Trainer
-        >>> from pytorch_lightning.loggers import TensorBoardLogger
-        >>> logger = TensorBoardLogger("tb_logs", name="my_model")
-        >>> trainer = Trainer(logger=logger)
+
+    .. code-block:: python
```
Comment (Borda) [inline pytorch_lightning/loggers/tensorboard.py:50]:
```suggestion
    .. testcode::
```

File: pytorch_lightning/loggers/tensorboard.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/loggers/tensorboard.py]: well, the difference is in `:` and `::`
with `:` it is recognized as a section title
with `::` it is recognized as caption followed by code
https://docs.typo3.org/m/typo3/docs-how-to-document/master/en-us/WritingReST/Codeblocks.html

**Review by Borda — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: docs/source/extensions/logging.rst [hunk 3]
```diff
@@ -294,6 +294,9 @@ Some loggers also allow logging the hyperparams used in the experiment. For inst
 when using the TestTubeLogger or the TensorBoardLogger, all hyperparams will show
 in the `hparams tab <https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_hparams>`_.
 
+.. note::
+    If you want to track a metric in the tensorboard hparams tab, log scalars to the key ``hp_metric``. If tracking multiple metrics, initialize ``TensorBoardLogger`` with ``default_hp_metric=False`` and call ``log_hyperparams`` only once with your metric keys and initial values. Subsequent updates can simply be logged to the metric keys.
```
Comment (tchaton) [inline docs/source/extensions/logging.rst:298]: Can you provide a code example ?

Comment (s-rog) [reply docs/source/extensions/logging.rst]: Done!

**Review by akihironitta — APPROVED** [review]
